src/app/stelliance-connect-widget/stelliance-connect-widget.component.ts

```diff
@@ -34,25 +35,65 @@ export class StellianceConnectWidgetComponent implements OnInit {
+      name: 'Conformitee',
+      clickCount: 0,
+    },
+    {
+      id: 'viaco2',
+      logo: 'https://c.smartrecruiters.com/sr-careersite-image-prod-dc5/5ed53051e3261858ae571e66/0549d054-280d-44de-86d1-0f9b9635f6cf?r=s3-eu-central-1',
+      name: 'Viaco',
+      clickCount: 0,
+    },
+    {
+      id: 'sis2',
+      logo: 'https://uploads-ssl.webflow.com/602be30475e9ea3cc47ec886/60f7dec17298039a8036b7cc_images-24-06-2021-09-33-665.png',
+      name: 'Sis ID',
+      clickCount: 0,
+    },
   ];
 
+  widgetAppsFavoritesLimit = defaultMinClickToFavorite;
+  public widgetAppsFavorites: StellianceConnectWidgetConfig[] = [];
+  public widgetAppsNotFavorites: StellianceConnectWidgetConfig[] = [];
+
   ngOnInit(): void {
     console.log('stelliance widget is initialized');
 
-    this.widgetApps.forEach((widgetApp) => {
-      widgetApp.clickCount = Number(localStorage.getItem(`${STELLIANCE_CONNECT_LINKS_COUNTER}_${widgetApp.id}`)) || 0;
-    });
-
-    this.widgetApps.sort(this.sortByMostClickedApps);
+    this.organizeWidgets();
   }
 
   navigateTo(widgetApp: StellianceConnectWidgetConfig) {
     this.storeApplicationClicks(widgetApp);
 
-    // navigate to app URL
+    // Active to organize widgets dynamically
+    this.organizeWidgets();
+
+    // Navigate to app URL
     // TODO
   }
 
+  private organizeWidgets() {
+    this.widgetApps.forEach((widgetApp) => {
+      widgetApp.clickCount = Number(localStorage.getItem(`${STELLIANCE_CONNECT_LINKS_COUNTER}_${widgetApp.id}`)) || 0;
+    });
+
+    this.widgetAppsNotFavorites = this.widgetApps
+      .filter((widgetApp) => widgetApp.clickCount < this.widgetAppsFavoritesLimit)
+      .sort(this.sortByMostClickedApps);
+    this.widgetAppsFavorites = this.widgetApps
+      .filter((widgetApp) => widgetApp.clickCount >= this.widgetAppsFavoritesLimit)
+      .sort(this.sortByMostClickedApps);
+  }
```

Comment:
A tester ca doit fonctionner et ca t'évite d'avoir à itérer 2 fois la liste des widget apps :
```
this.widgetApps
  .forEach((widgetApp) => 
    (widgetApp.clickCount < this.widgetAppsFavoritesLimit ? this.widgetAppsNotFavorites : this.widgetAppsFavorites).push(widgetApp)
  )
 .sort(this.sortByMostClickedApps)
```

Reply:
Je me demande meme si en train initialement `this.widgetApps` ca ne fonctionne pas à tester:

```
this.widgetAppsNotFavorites.sort(widgetApps);
this.widgetApps
  .forEach((widgetApp) => 
    (widgetApp.clickCount < this.widgetAppsFavoritesLimit ? this.widgetAppsNotFavorites : this.widgetAppsFavorites).push(widgetApp)
  );

Reply:
ok